library/src/main/java/org/apache/apex/malhar/lib/fs/s3/S3StringOutputModule.java

```diff
@@ -0,0 +1,243 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.apex.malhar.lib.fs.s3;
+
+import javax.validation.constraints.Min;
+
+import org.apache.apex.malhar.lib.fs.GenericFileOutputOperator;
+import org.apache.hadoop.conf.Configuration;
+
+import com.esotericsoftware.kryo.NotNull;
+
+import com.datatorrent.api.DAG;
+import com.datatorrent.api.Module;
+
+@org.apache.hadoop.classification.InterfaceStability.Evolving
+public class S3StringOutputModule implements Module
+{
+  public final transient ProxyInputPort<String> input = new ProxyInputPort<String>();
+
+  /**
+   * AWS access key
+   */
+  @NotNull
+  private String accessKey;
+  /**
+   * AWS secret access key
+   */
+  @NotNull
+  private String secretAccessKey;
+
+  /**
+   * S3 End point
+   */
+  private String endPoint;
+  /**
+   * Name of the bucket in which to upload the files
+   */
+  @NotNull
+  private String bucketName;
+
+  /**
+   * Path of the output directory. Relative path of the files copied will be
+   * maintained w.r.t. source directory and output directory
+   */
+  @NotNull
+  private String outputDirectoryPath;
+
+  /**
+   * Max number of idle windows for which no new data is added to current part
+   * file. Part file will be finalized after these many idle windows after last
+   * new data.
+   */
+  private long maxIdleWindows = 30;
+
+  /**
+   * The maximum length in bytes of a rolling file. The default value of this is 1MB.
+   */
+  @Min(1)
+  protected Long maxLength = 128 * 1024 * 1024L;
+
+  protected static final String S3_INTERMEDIATE_DIR = "S3_INTERMEDIATE_DIR";
+
+  @Override
+  public void populateDAG(DAG dag, Configuration conf)
+  {
+    S3CompactionOperator<String> s3compaction = dag.addOperator("S3Compaction", new S3CompactionOperator());
+    s3compaction.setConverter(new GenericFileOutputOperator.StringToBytesConverter());
```

Comment:
Most of the code in S3StringOutputModule, S3BytesOutputModule is redundant. Please merge both the modules or create AbstractOutputModule which has common code.